packages/manager/cypress/e2e/core/cloudpulse/timerange-verification.spec.ts

```diff
@@ -0,0 +1,519 @@
+   * hour, and minute. The function also provides individual date components such as day, hour,
+   * minute, month, and AM/PM.
+   *
+   * @param {number} daysOffset - The number of days to adjust from the current date. Positive
+   *                               values give a future date, negative values give a past date.
+   * @param {number} hour - The hour to set for the resulting date (0-23).
+   * @param {number} [minute=0] - The minute to set for the resulting date (0-59). Defaults to 0.
+   *
+   * @returns {Object} - Returns an object containing:
+   *   - `actualDate`: The formatted date and time in IST (YYYY-MM-DD HH:mm).
+   *   - `day`: The day of the month as a number.
+   *   - `hour`: The hour in the 24-hour format as a number.
+   *   - `minute`: The minute of the hour as a number.
+   *   - `month`: The month of the year as a number.
+   */
+  const getDateRangeInGMT = (
+    daysOffset: number,
+    hour: number,
+    minute: number = 0
+  ) => {
+    const now = DateTime.now().setZone('GMT'); // Set the timezone to GMT
+    const targetDate = now
+      .startOf('month')
+      .plus({ days: daysOffset })
+      .set({ hour, minute });
+
+    const actualDate = targetDate.toFormat('yyyy-LL-dd HH:mm'); // Format in GMT
+    return {
+      actualDate,
+      day: targetDate.day,
+      hour: targetDate.hour,
+      minute: targetDate.minute,
+      month: targetDate.month,
+    };
+  };
+
+  it('Implement and validate the functionality of the custom date and time picker for selecting a specific date and time range', () => {
+    // Calculates start and end dates in GMT using `getDateRangeInGMT` for testing date and time ranges.
+    const {
+      actualDate: startActualDate,
+      day: startDay,
+      hour: startHour,
+      minute: startMinute,
+    } = getDateRangeInGMT(0, 12, 15);
+    const {
+      actualDate: endActualDate,
+      day: endDay,
+      hour: endHour,
+      minute: endMinute,
+    } = getDateRangeInGMT(currentDate.getDate(), 12, 15);
+
+    // Select "Custom" from the "Time Range" dropdown
+    ui.autocomplete
+      .findByLabel('Time Range')
+      .scrollIntoView()
+      .should('be.visible')
+      .type('Custom');
+
+    // Select "Custom" from the autocomplete dropdown
+    ui.autocompletePopper.findByTitle('Custom').should('be.visible').click();
+
+    // Click on "Select Start Date" input field
+    cy.findByPlaceholderText('Select Start Date').should('be.visible').click();
+
+    // Select the start date from the calendar
+    cy.findByRole('gridcell', { name: startDay.toString() })
+      .should('be.visible')
+      .click();
+
+    // Clicks the button closest to the Clock Icon, bypassing any visible state with `force: true`.
+    cy.get('[data-testid="ClockIcon"]')
+      .closest('button')
+      .click({ force: true });
+
+    // Selects the start hour, minute, and meridiem (AM/PM) in the time picker.
+
+    cy.get('[aria-label="Select hours"]')
```

Comment:
```suggestion
    cy.findByLabelText('Select hours')
```
Everywhere where we're calling `cy.get('[aria-label="..."]')` can be replaced with `cy.findByLabelText('...')`